frontend/src/Frontend/UI/Dialogs/Send.hs

```diff
@@ -335,48 +335,62 @@ sendConfig model initData = Workflow $ do
           displayImmediateFeedback (updated decoded) cannotBeReceiverMsg
             $ either (const False) (\ka -> _txBuilder_accountName ka == fromName && _txBuilder_chainId ka == fromChain)
 
-          (_, amount, _) <- mkLabeledInput True "Amount" uiGasPriceInputField
-            (def & inputElementConfig_initialValue .~ (maybe "" tshow mInitAmount))
-            -- We're only interested in the decimal of the gas price
-            <&> over (_2 . mapped . mapped) (\(GasPrice (ParsedDecimal d)) -> d)
+          let
+            balance = _account_status fromAcc ^? _AccountStatus_Exists . accountDetails_balance . to unAccountBalance
+
+            gasInputWithMaxButton cfg = mdo
+              let projectAmount = over (_2 . mapped . mapped) (\(GasPrice (ParsedDecimal d)) -> d)
+                  attrs = ffor useEntireBalance $ \u -> "disabled" =: ("disabled" <$ u)
+              (_, inputAmount, _) <- fmap projectAmount $ uiGasPriceInputField $ cfg
+                & inputElementConfig_elementConfig . elementConfig_modifyAttributes .~ updated attrs
+              useEntireBalance <- case balance of
+                Nothing -> pure $ pure Nothing
+                Just b -> fmap (\v -> b <$ guard v) . _checkbox_value <$> uiCheckbox "input-max-toggle" False def (text "Max")
```

Comment:
Kadena are the bosses here, but it really feels like this should be called "Drain" or something. That's the word everyone keeps throwing around in demos. But maybe it makes complete sense if you put the amount into the input when max is clicked.